lib/ruby_llm/chat.rb

```diff
@@ -92,16 +95,22 @@ def each(&)
       messages.each(&)
     end
 
+    def cache_prompts(system: false, user: false, tools: false)
+      @cache_prompts = { system: system, user: user, tools: tools }
+      self
+    end
+
     def complete(&)
       @on[:new_message]&.call
-      response = @provider.complete(
-        messages,
+      params = CompletionParams.new(
+        messages: messages,
         tools: @tools,
         temperature: @temperature,
         model: @model.id,
-        connection: @connection,
-        &
+        cache_prompts: @cache_prompts.dup,
+        connection: @connection
       )
+      response = @provider.complete(params, &)
```

Comment:
I don't feel like it's necessary to make a new `CompletionParams` class. Let's _remove_ complexity.

Reply:
@tpaulshippy yes I'd say so, just add a disable comment.